docs/persister.md

```diff
@@ -0,0 +1,56 @@
+# Scorch Index Memory and File Management
+
+## Memory Management
+
+When data is indexed in Scorch — using either the `index.Index()` or `index.Batch()` API — it is added as part of an in-memory "segment". Memory management in Scorch indexing mainly relates to handling these in-memory segments during workloads that involve inserts or updates. 
+
+In scenarios with a continuous stream of incoming data, a large number of in-memory segments can accumulate over time. This is where the persister component comes into play—its job is to flush these in-memory segments to disk.
+
+Starting with v2.5.0, Scorch supports parallel flushing of in-memory segments to disk, where the persister checks the total in-memory data and distributes the flush across multiple workers. This feature is disabled by default and can be enabled using two configuration options:
+
+- `NumPersisterWorkers`: This factor decides how many maximum workers can be spawned to flush out the in-memory segments. Each worker is supposed to work on a disjoint subset of segments, merge them, and flush them out to the disk. By default the persister deploys only one worker.
+- `MaxSizeInMemoryMergePerWorker`: This config decides what's the maximum amount of input data in bytes a single worker can work upon. By default this value is equal to 0 which means that this config is disabled and the worker tries to merge all the data in one shot.
```

Comment:
Should we add that it's imperative that the user set this config if `NumPersisterWorkers > 1` ?